crates/ruff/src/rules/pylint/rules/else_if_used.rs

```diff
@@ -0,0 +1,45 @@
+use ruff_macros::{define_violation, derive_message_formats};
+use rustpython_parser::ast::{Stmt, StmtKind};
+
+use crate::ast::helpers::identifier_range;
+use crate::registry::Diagnostic;
+use crate::source_code::Locator;
+use crate::violation::Violation;
+
+define_violation!(
+    pub struct ElseIfUsed {}
+);
+
+impl Violation for ElseIfUsed {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Consider using \"elif\" instead of \"else\" then \"if\" to remove one indentation level")
+    }
+}
+
+/// PLR5501
+pub fn else_if_used(
+    stmt: &Stmt,
+    body: &[Stmt],
+    _orelse: &[Stmt],
+    locator: &Locator,
+) -> Option<Diagnostic> {
+    if let StmtKind::If { orelse, .. } = &stmt.node {
+        // If the body contains more than just the orelse, can't apply.
+        if body.len() == 1 {
```

Comment:
Should this be `orelse.len() == 1`? I thought we'd be looking for `orelse.len() == 1 && matches!(orelse[0].node, StmtKind::If { .. })`, which typically indicates an `elif` (or an `else: if:` in this case, which we'd have to differentiate somehow).

Reply:
Ah, here `stmt` is the `if` in the `else: if: ...`. I'd assumed `stmt` was the outer `if` in:

```py
if ...:
  pass
else:
  if:
    ...
```

Reply:
Hmm, I think we _do_ need to structure this such that it looks at this statement:

```py
if ...:
  pass
else:
  if:
    ...
```

Rather than the nested `if`. Otherwise, how can you tell if you're in an `else` or not?

Reply:
Yeah, this is more complicated than I thought, namely because you don't know where the else is.
Pylint does have separate `elif` tokens which makes it easier...the only way I can think of doing it might be a bit janky where you use the locator to see if 'elif' is in the line or something like that.

Reply:
So this can be done via the text method...if that ok? It feels a bit...dirty...

Reply:
Yeah it's ok. The alternative would be to do it by lexing and looking at the token stream. But this is effectively equivalent. There's no way to detect this AFAIK from the AST alone.